src/main/java/uk/gov/hmcts/reform/pip/rules/repository/InMemoryRepository.java

```diff
@@ -1,75 +1,75 @@
+
+        ObjectMapper om = new ObjectMapper().setDateFormat(new SimpleDateFormat("dd/MM/yyyy"));
+
+        InputStream fileHearings = getClass().getClassLoader().getResourceAsStream("hearingsList.json");
+        //File fileHearings = new File(mockPath + "hearingsList.json");
+        Hearing[] list = om.readValue(fileHearings, Hearing[].class);
+        this.listHearings = Arrays.asList(list);
+
+        InputStream fileCourts = getClass().getClassLoader().getResourceAsStream("courtsAndHearingsCount.json");
+        Court[] courts = om.readValue(fileCourts, Court[].class);
+
+        this.listCourts = Arrays.asList(courts);
+
+    }
+
 
     /**
-     * Method to retrieve a publication using it's ID.
-     * @param publicationId The ID of the publication.
-     * @return The publication to return, if found.
+     * Method to retrieve a court hearings using court ID.
+     * @param courtId The ID of the court.
+     * @return The court hearings to return, if found.
      */
-    public Optional<Publication> getPublication(Integer publicationId) {
-
-        if (publicationId.equals(AVAILABLE_PUBLICATION)) {
-            CourtHearings courtHearings = new CourtHearings();
-            courtHearings.setCourtId(1);
-            courtHearings.setName("This is a court");
-
-            Hearing hearing = new Hearing();
-            hearing.setHearingId(1);
-            hearing.setCourtId(1);
-            hearing.setDate(LocalDate.now());
-            hearing.setTime(LocalTime.now());
-            hearing.setJudge("This is a judge");
-            hearing.setPlatform("This is a platform");
-            hearing.setCaseNumber("This is a case number");
-            hearing.setCaseName("This is a case name");
-
-            Hearing otherHearing = new Hearing();
-            otherHearing.setHearingId(2);
-            otherHearing.setCourtId(1);
-            otherHearing.setDate(LocalDate.now());
-            otherHearing.setTime(LocalTime.now());
-            otherHearing.setJudge("This is a judge");
-            otherHearing.setPlatform("This is a platform");
-            otherHearing.setCaseNumber("This is a case number");
-            otherHearing.setCaseName("This is a case name");
-
-            List<Hearing> hearings = new ArrayList<>();
-            hearings.add(hearing);
-            hearings.add(otherHearing);
-            courtHearings.setHearingList(hearings);
-
-            Publication publication = new Publication();
-            publication.setPublicationId(1);
-
-            List<CourtHearings> courtHearingsList = new ArrayList<>();
-            courtHearingsList.add(courtHearings);
-
-            publication.setCourtHearingsList(new ArrayList<>(courtHearingsList));
-            return Optional.of(publication);
+    public Optional<Court> getCourtHearings(Integer courtId) {
+        Court court = this.listCourts.stream()
+            .filter(c -> courtId.equals(c.getCourtId()))
+            .findAny()
+            .orElse(null);
```

Comment:
.findAny returns an Optional, so I don't think we need the orElse (and the Optional.empty() on line 61 can just return the result)

Reply:
code refactored